### PR TITLE
[Identify] Make Anonymous ID customisable.

### PR DESCRIPTION
**Pull request by alloy**
This is a first stab at making the Anonymous ID customisable, as asked about in #482. Some more background can be found in [this PR](https://github.com/artsy/eigen/pull/958) on our app.

This is very much a work in progress, I did the bare minimum to get _what we needed_ to work, so now I’m looking for feedback on:

* Is adding this feature conceptually ok?
* Would you prefer I refactor the `-getAnonymousId:` code into e.g. the getter of the `anonymousId` property, or what is the style you’d prefer?
* General code-style

### CODE REVIEW

**Comment by f2prateek** [comment]
@alloy Thanks! This looks great, and should work as expected. I don't really have a preference for the style of the `-getAnonymousId:` method, unless you have a compelling reason to pick one or the other. General code style looks fine as well, I'll run it through our code formatter after merging.

My only question is if there is a better to expose this than hiding it in under the `options` parameter.
```objc
[[SEGAnalytics sharedAnalytics] identify:nil
                                           traits:@{ "b" : "c" }
                                          options:@{ @"anonymousId":@"d" }];

vs.

[[SEGAnalytics sharedAnalytics] setAnonymousID:@"d"];
[[SEGAnalytics sharedAnalytics] identify:nil traits:@{ "b" : "c" }];
```

**Comment by alloy** [comment]
Nope, my only reason for asking it is purely to conform to your code base design ideas :)

Regarding exposing it, I’m fine with either, but maybe for consistency a method like 
`-[SEGAnalytics identify:anonymousID:traits:options:]` would be better? Whatever you choose, let me know and I’ll wrap this patch up.

**Comment by mcstoufer** [comment]
It looks like any track() calls made before the setAnonymousId() and  identify() calls will miss using the intended custom anonymousId. Would it be up to the developer to implicitly understand this and they take the responsibility of setting the ID when they need it in the flow?

**Comment by f2prateek** [comment]
@mcstoufer Yes, the current approach requires that if you want to use a custom anonymousID, you *must* call identify with the anonymous ID first (typically right after initializing the analytics client). Subsequent calls don't need to provide the anonymous ID.

**Comment by alloy** [comment]
@mcstoufer Indeed. I’m guessing that that’s what most people already do anyways, or at least that it’s safe assumption.

**Comment by f2prateek** [comment]
@alloy I think we can merge this change. Could you add a note in the [docs](https://github.com/segmentio/analytics-ios/blob/master/Pod/Classes/SEGAnalytics.h#L80-L92) to indicate how to use the `options` parameter to set a custom anonymous id?

File: Pod/Classes/SEGAnalytics.h [hunk 1]
```diff
@@ -88,6 +88,8 @@
 
  @param traits        A dictionary of traits you know about the user. Things like: email, name, plan, etc.
 
+ @param options       A dictionary of options, such as the `@"anonymousId"` key. If no anonymous ID is specified one will be generated for you.
+
```
Comment (alloy) [inline Pod/Classes/SEGAnalytics.h]: @f2prateek Is this good?

**Comment by alloy** [comment]
Updated.

File: Pod/Classes/SEGAnalytics.h (hunk 1 above)
Comment (f2prateek) [inline Pod/Classes/SEGAnalytics.h]: Yup! We might want to clarify in the future that not bundled integrations will respect this, but we'll see if this confuses users first.

Comment (alloy) [inline Pod/Classes/SEGAnalytics.h]: :+1:

**Pull request merged**